src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -176,11 +193,15 @@ private void removeDeadDataNodes() {
     return getHashMap(DATANODES_KEY);
   }
 
-  public Collection<String> getAllTaskIds() {
-    HashMap<String, String> allTasksIds = getJournalNodes();
-    allTasksIds.putAll(getNameNodes());
-    allTasksIds.putAll(getDataNodes());
-    return allTasksIds.values();
+  public Set<String> getAllTaskIds() {
+    Set<String> allTaskIds = new HashSet<String>();
+    Collection<String> journalNodes = getJournalNodes().values();
+    Collection<String> nameNodes = getNameNodes().values();
+    Collection<String> dataNodes = getDataNodes().values();
+    allTaskIds.addAll(journalNodes);
+    allTaskIds.addAll(nameNodes);
```

Comment:
I guess the ZK tasks are bundled in with the NNs? They don't get their own taskIds?